src/main/java/com/lgtm/easymoney/services/impl/FeedServiceImpl.java

```diff
@@ -0,0 +1,117 @@
+package com.lgtm.easymoney.services.impl;
+
+import com.lgtm.easymoney.enums.TransactionStatus;
+import com.lgtm.easymoney.enums.UserType;
+import com.lgtm.easymoney.exceptions.InapplicableOperationException;
+import com.lgtm.easymoney.exceptions.InvalidUpdateException;
+import com.lgtm.easymoney.exceptions.ResourceNotFoundException;
+import com.lgtm.easymoney.models.Friendship;
+import com.lgtm.easymoney.models.Transaction;
+import com.lgtm.easymoney.models.User;
+import com.lgtm.easymoney.payload.FeedActivityRsp;
+import com.lgtm.easymoney.payload.FeedRsp;
+import com.lgtm.easymoney.payload.FriendshipReq;
+import com.lgtm.easymoney.payload.ProfileRsp;
+import com.lgtm.easymoney.payload.ProfilesRsp;
+import com.lgtm.easymoney.payload.TransactionRsp;
+import com.lgtm.easymoney.repositories.FriendshipRepository;
+import com.lgtm.easymoney.services.FeedService;
+import com.lgtm.easymoney.services.FriendService;
+import com.lgtm.easymoney.services.TransactionService;
+import com.lgtm.easymoney.services.UserService;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+import net.sf.saxon.trans.SymbolicName;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+
+/**
+ * service for friendship.
+ * TODO after we have auth, move the inapplicable op checks to a HandlerInterceptor instead
+ */
+@Service
+@Transactional(rollbackFor = Exception.class)  // required for delete
+public class FeedServiceImpl implements FeedService {
+  private final UserService userService;
+  private final FriendService friendService;
+  private final TransactionService transactionService;
+
+  /**
+   * feed service for getting user's feed activity.
+   *
+   * @param userService userservice
+   * @param friendService friend service
+   * @param transactionService trans service
+   */
+  @Autowired
+  public FeedServiceImpl(UserService userService, FriendService friendService,
+                         TransactionService transactionService) {
+    this.userService = userService;
+    this.friendService = friendService;
+    this.transactionService = transactionService;
+  }
+
+  @Override
+  public List<Transaction> getFeedByUser(User u) {
+    // only return 20 latest and valid transaction
```

Comment:
also in next pr

Reply:
actually we can talk about this, i think feed should be time-sensitive and 20 sounds reasonable?